main.py

```diff
@@ -675,31 +687,40 @@ def post(self):
             return
 
-        if 0: # self.request.get('action') == "emailblast":
-            unverified_row_ids = {}
-            query = db.GqlQuery("SELECT * FROM User")
-            # Example of unverified_row_id:
-            # {'unconfirmed_account@eastsideprep.org': '5170986005561344',
-            # 'email_with_some_confirmed_and_some_not@eastsideprep.org': None}
-
-            for query_result in query:
-                if not query_result.verified:
-                    if not query_result.email in unverified_row_ids:
-                        id_key = str(query_result.key().id())
-                        unverified_row_ids[query_result.email] = id_key
-                else: # query_result.verified
-                    unverified_row_ids[query_result.email] = "verified"
-
-            for attribute in unverified_row_ids:
-                if unverified_row_ids[attribute] != "verified":
-                    logging.info("Sending " + query_result.email + " a verification email")
-                    try:
-                        error = self.send_confirmation_email(query_result.email, id_key)
-                    except: # If email is ficticious or something else went wrong
-                        logging.error("Attempted to send an email to " + query_result.email + ", was unsuccessful")
-
-        #elif self.request.get('action') == "cleanup":
-            #query = db.GqlQuery("SELECT * FROM User")
+        if self.request.get('action') == "emailblast":
+            self.send_email_blast()
+        elif self.request.get('action') == "cleanup":
+            self.clean_up_db()
+
+    def send_email_blast(self):
+        unverified_row_ids = {}
+        # Example of what unverified_row_ids will be:
+        # {'unconfirmed_account@eastsideprep.org': '5170986005561344',
+        # 'email_with_some_confirmed_and_some_not@eastsideprep.org': None}
+
+        # Generate unverified_row_ids
+        for query_result in query:
+            if not query_result.verified:
+                if not query_result.email in unverified_row_ids:
+                    id_key = str(query_result.key().id())
+                    unverified_row_ids[query_result.email] = id_key
+            else: # query_result.verified
+                unverified_row_ids[query_result.email] = "verified"
+
+        # For each email with no verified properties, send them the verification email
+        for attribute in unverified_row_ids:
+            if unverified_row_ids[attribute] != "verified":
+                logging.info("Sending " + query_result.email + " a verification email")
+                try:
+                    error = self.send_confirmation_email(query_result.email, id_key)
+                except: # If email is ficticious or something else went wrong
+                    logging.error("Attempted to send an email to " + query_result.email + ", was unsuccessful")
+
+    def clean_up_db(self):
+        verification = self.get_db()
+        for email in verification:
```

Comment:
consider using temporary vars here to make the code simpler
verified = verification['email']['verified']
unverified = verification['email']['unverified']
if len(verified) == 1 ...
  db.delete(unverified)
db.delete(rows['unverified